maven-plugin/src/main/java/org/openmrs/maven/plugins/Setup.java

```diff
@@ -186,6 +195,31 @@ private DistroProperties resolveDistroProperties(Server server) throws MojoExecu
 					}
 					platformMode = false;
 					break;
+				case O3_Distribution:
+					wizard.promptForO3RefAppVersionIfMissing(server, versionsHelper);
+					Artifact artifact = new Artifact(server.getDistroArtifactId(), server.getVersion(),
+							server.getDistroGroupId(), "zip");
+					Properties frontendProperties;
+					if (server.getVersion().equals(versionsHelper.getLatestSnapshotVersion(artifact))) {
+						frontendProperties = PropertiesUtils.getFrontendPropertiesFromSpaConfigUrl(
+								"https://raw.githubusercontent.com/openmrs/openmrs-distro-referenceapplication/main/frontend/spa-build-config.json");
+					} else {
+						frontendProperties = PropertiesUtils.getFrontendPropertiesFromSpaConfigUrl(
+								"https://raw.githubusercontent.com/openmrs/openmrs-distro-referenceapplication/"+ server.getVersion() +"/frontend/spa-build-config.json");
+					}
+					Properties configurationProperties = PropertiesUtils.getConfigurationProperty(artifact);
```

Comment:
Do we need a separate properties object for the above? Since we are just generating it and putting it in that properties file, can't we just add it to the backendProperties?

Reply:
distro.properties file in `org.openmrs.distro:referenceapplication-distro` does not have the configurations. So we have to manually add those properties. We will only get config.referenceapplication-distro if users are providing a custom distro.properties.